Create petpal app layout and login function

lujorryn opened the pull request:
1. Setup global components that can be used on multiple pages
  - Button
  - Card, CardWrapper
  - FormField, Input
2. Create Header that changes based on user role and Footer
  - Nav 
      - index: navbar that's on the right hand side of the header 
      - AccountNav: Dropdown menu that links to profile, messages, logout 
      - LinkItem: represents each navlink 
      - NotificationNav: Notif dropdown, implementation TBC
  - Header: Includes Logo that links to home and Nav that links to other pages
  - SearchModal: links to WIP search page
3. Implement login functionality via context, returns token, userId, role and login logout functions.
4. Implement fully functional login page
5. Setup skeleton pages for the entire app

lujorryn committed:
Create petpal app layout and login function

1. Setup global components that can be used on multiple pages
  - Button
  - Card, CardWrapper
  - FormField, Input
2. Create Header that changes based on user role and Footer
  - Nav
    - index: navbar that's on the right hand side of the header
    - AccountNav: Dropdown menu that links to profile, messages, logout
    - LinkItem: represents each navlink
    - NotificationNav: Notif dropdown, implementation TBC
  - Header: Includes Logo that links to home and Nav that links to other
    pages
  - SearchModal: links to WIP search page
3. Implement login functionality via context, returns token, userId,
role and login logout functions.
4. Implement fully functional login page
5. Setup skeleton pages for the entire app

lujorryn committed:
Example of checking for authentication for a page

Implement sample useAuth for the shelters page.

iamyanbo commented on frontend/src/context/AuthContext.js:
```diff
@@ -0,0 +1,70 @@
+    localStorage.setItem('accessToken', accessToken)
+    localStorage.setItem('refreshToken', refreshToken)
+
+    checkUserRole(accessToken)
```
await async function

lujorryn replied:
Fixed in [f6a47fc](https://github.com/lujorryn/PetPal-G0346/pull/22/commits/f6a47fc78978550def9d1e1674e9d96960f35e95)

iamyanbo commented on frontend/src/context/AuthContext.js:
```diff
@@ -0,0 +1,70 @@
+        }
+      })
+
+      if (!res.ok) return setRole('seeker')
```
I get that this will set a role, but it doesn't keep the state on redirect or refresh, this should be stored in localstorage as well as in your navbar it can be null whe it checks for the role when the user is logged in. For example try logging in and refreshing the page

lujorryn replied:
Fixed in [f6a47fc](https://github.com/lujorryn/PetPal-G0346/pull/22/commits/f6a47fc78978550def9d1e1674e9d96960f35e95)

iamyanbo commented on frontend/src/components/ui/Nav/AccountNav.jsx:
```diff
@@ -0,0 +1,41 @@
+        onClick={() => {
+          onClick()
+          logout() 
+        }}
```
This is correctly logging the user out but the navigate to /login doesn't work for some reason, I would just add a to='messages' as well

lujorryn replied:
Fixed in [f6a47fc](https://github.com/lujorryn/PetPal-G0346/pull/22/commits/f6a47fc78978550def9d1e1674e9d96960f35e95)
Not really sure why navigate doesn't work either. I added `to='/login'` so it should be working now.

lujorryn committed:
Fix small issues in app setup

1. await checkUserRole because it's an async function
2. add 'role', 'userId' to localStorage so it still works after refresh
3. have the logout button link to the login page
4. Add id attribute to Input componenet in case different components
have the same name (optional: id still defaults to name)

lujorryn commented:
Added userId to localStorage and fixed a duplicate id problem for the Input component in [force-push](https://github.com/lujorryn/PetPal-G0346/compare/f6a47fc78978550def9d1e1674e9d96960f35e95..aef739c8a794aef054329b1a51ea3337c9585fbc)
